R/nm-tables.R

```diff
@@ -0,0 +1,131 @@
+
+#' Read all tables and input data
+#'
+#' Reads in the input data set and all table output files from a NONMEM run.
+#' @return A named list of tibbles. The first element will always be named
+#'   `data` and will contain the input data set. Subsequent elements will be
+#'   named for the file from which they were loaded, with `get_model_id(.mod)`
+#'   (and `.`) removed from the beginning and end, if present. For example, a
+#'   model named `001.ctl` that generated a table named `001.tab` will have the
+#'   relevant element named `tab`.
+#' @param .mod Either a `bbi_nonmem_model` or `bbi_nonmem_summary` object
+#' @param .files Character vector of file paths to table files to read in.
+#'   Defaults to calling [nm_table_files()] on `.mod`, which will parse all file
+#'   names from `$TABLE` blocks in the control stream.
+#' @importFrom purrr compact map_chr
+#' @importFrom stringr str_replace
+#' @export
+nm_tables <- function(
+  .mod,
+  .files = nm_table_files(.mod)
+) {
+  check_model_object(.mod, c(NM_MOD_CLASS, NM_SUM_CLASS))
+  checkmate::assert_character(.files)
+
+  # make paths absolute
+  .files <- map_chr(.files, ~{
+    if (fs::is_absolute_path(.x)) {
+      return(.x)
+    } else {
+      return(file.path(get_output_dir(.mod), .x))
+    }
+  })
+
+  # read in input data
+  res <- list(
+    data = nm_data(.mod)
+  )
+
+  # build names for table elements
+  .n <- .files %>%
+    basename() %>%
+    str_replace(glue("^{get_model_id(.mod)}"), "") %>%
+    str_replace(glue("{get_model_id(.mod)}$"), "") %>%
```

Comment:
* I think the model ID return value is arbitrary enough input that it should be protected with `\\Q...\\E`.  

 * I wasn't able to figure out the scenario where the `id$` case applies.  That case doesn't seem to be covered by the tests because those pass when I remove line 43.

 * Assuming that second call is needed: mostly a matter of taste that you can ignore if you don't agree, but I think it'd be good to lift the two `get_model_id` calls into one upstream one stored in a variable.

Reply:
the second one is necessary, but a good point that we never test it. I'll add one.